Draygon152/ICS167-Project
# Finalizing game loop

Testing instructions

You should be able to create waves of enemies by creating a new wave in the Editor then filling it with enemies and spawn points. The two lists are parallel, so the ith enemy in the prefab list will spawn at the ith spawn point in the spawn point list. If the size of the two list are not equal an exception should be thrown by the wave's awake function. Note that the spawn points need to be an assets to be able to put it into a scriptable object.

Once a wave is created you be able to load an number of them into the GameManager. The order they are in the Game Mangager waves list is the order in which the waves will be spawned. When the number of enemies remaining drops to the const in the GameManager (set to 2 by default) the next wave will be spawned. If we are on the last wave then the Game Manager will let all the enemies die before triggering the PlayerDefeat Event.
